components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -84,6 +107,7 @@ public StorageTxImpl(final BlobTx blobTx,
     this.graphTx = checkNotNull(graphTx);
     this.bucketId = checkNotNull(bucketId);
     this.writePolicy = writePolicy;
+    this.links = Lists.newArrayList();
```

Comment:
I suppose this implies that the links aren't traversable during the transaction they're being created.